src/pip/_internal/operations/install/wheel.py

```diff
@@ -381,8 +380,7 @@ def clobber(
                     continue
                 elif (
                     is_base and
-                    s.endswith('.dist-info') and
-                    canonicalize_name(s).startswith(canonicalize_name(name))
+                    s.endswith('.dist-info')
```

Comment:
On the other end, it would now mean that a wheel can contain a single unrelated dist-info dir.

Reply:
I'll add an assertion below.

Reply:
Added [here](https://github.com/pypa/pip/pull/7494/files#diff-81eaeaa2196a8c5382958f2d9f22b593R449-R456).